### PR TITLE
controller: Clean up the instance manager when the node is disconnected

### PR DESCRIPTION
**Pull request by shuo-wu**
Fix the 1st scenario in longhorn/longhorn#1545

### CODE REVIEW

**Comment by yasker** [comment]
Need an explanation for the PR.

File: controller/instance_manager_controller.go [hunk 1]
```diff
@@ -324,6 +327,10 @@ func (imc *InstanceManagerController) syncInstanceManager(key string) (err error
 		if im.Status.CurrentState != types.InstanceManagerStateError {
 			im.Status.CurrentState = types.InstanceManagerStateStopped
 		}
+	} else if pod.DeletionTimestamp != nil {
+		if im.Status.CurrentState != types.InstanceManagerStateError {
+			im.Status.CurrentState = types.InstanceManagerStateError
+		}
 	} else {
 		// TODO: Will remove this reference kind correcting after all Longhorn components having used the new kinds
```
Comment (joshimoo) [inline controller/instance_manager_controller.go:335]: Are these TODO's still relevant?

File: controller/instance_manager_controller.go [hunk 2]
```diff
@@ -324,6 +327,10 @@ func (imc *InstanceManagerController) syncInstanceManager(key string) (err error
 		if im.Status.CurrentState != types.InstanceManagerStateError {
 			im.Status.CurrentState = types.InstanceManagerStateStopped
 		}
+	} else if pod.DeletionTimestamp != nil {
```
Comment (joshimoo) [inline controller/instance_manager_controller.go:330]: can be combinded with above `pod == nil || pod.DeletionTimestamp != nil`

File: controller/instance_manager_controller.go (hunk 1 above)
Comment (shuo-wu) [reply controller/instance_manager_controller.go:335]: This TODO is not related to this issue. We can create a separate PR for that.

File: controller/instance_manager_controller.go (hunk 2 above)
Comment (shuo-wu) [reply controller/instance_manager_controller.go:330]: The IM object may be just created or the IM pod is just gone when `pod == nil`, then we don't need to update the IM status to `Error`. But there must be something wrong when `pod.DeletionTimestamp != nil` and the IM status should be `Error` in this case. Hence we cannot blindly combine these 2 cases.

**Pull request merged**